setup.py

```diff
@@ -1,19 +1,148 @@
 import os
 import sys
+import platform
+
+from distutils.core import setup, Extension 
 
 if os.name == 'nt':
     from distutils import msvc9compiler
     msvc9compiler.VERSION = 14.0 #Visual studio 2015
 
-if sys.version > '3':
-    setup_file = "setup_3.py"
+# Get the script directory.
+def get_script_dir():
+    path = os.path.abspath(sys.argv[0])
+    if os.path.isdir(path):
+        return path
+    elif os.path.isfile(path):
+        return os.path.dirname(path)
+
```

Comment:
suggestion for cci-submodule:
1. install python-git (sudo yum install python-pip3)
2. import git, subprocess
```
    import git
    import subprocess
```
3. cci-submodule update if not exists
def git(*args):
        return subprocess.check_call(['git'] + list(args))

```
if os.path.isfile("cci-src/configure") == 0:
        git ("submodule", "init")
        git ("submodule", "update")
```

Reply:
okay. i'll add the code. and add "gitpython" information in README.md. thank you